File: string_array/Best_Time_to_Buy_and_Sell_Stock.java
class Solution {
    public int maxProfit(int[] prices) {
    int low = prices[0];
    int best_sell = 0;
    int sell;
    for (int i = 0; i < prices.length; ++i) {
        if (prices[i] < low) {
            low = prices[i];
        }
        sell = prices[i] - low;
        if (sell > best_sell) {
            best_sell = sell;
        }
    }
    return best_sell;
    }
}

File: string_array/Best_Time_to_Buy_and_Sell_Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        low = prices[0]
        best_sell = 0
        sell = 0
        for i in prices:
            if i < low:
                low = i
            sell = i - low
            if sell > best_sell:
                best_sell = sell
        return best_sell